launch/launch/actions/execute_process.py

```diff
@@ -716,6 +735,31 @@ async def __execute_process(self, context: LaunchContext) -> None:
             self.__logger.error("process has died [pid {}, exit code {}, cmd '{}'].".format(
                 pid, returncode, ' '.join(cmd)
             ))
+            # respawn the process that abnormally died
+            if not context.is_shutdown and self.__respawn:
```

Comment:
@iuhilnehc-ynos hmm, I wonder if we should respawn if a shutdown event was sent explicitly (which is not the same as a full `launch` shutdown.)

Reply:
@hidmic 
Not very clear about this comment. Do you mean I need to add extra checking about **not self.__shutdown_received**?

Reply:
Correct. I'd think the `respawn` feature should apply to executables that die due to reasons external to `launch`.